qutebrowser/misc/backendproblem.py

```diff
@@ -397,6 +399,28 @@ def _check_backend_modules():
     raise utils.Unreachable
 
 
+def _handle_cache_nuking():
+    """Nuke the QtWebEngine cache if the Qt version changed.
+
+    WORKAROUND for https://bugreports.qt.io/browse/QTBUG-72532
+    """
+    if not configfiles.state.qt_version_changed:
+        return
+
+    # Only nuke the cache in cases where we know there are problems.
+    # Issues got introduced with Qt 5.12 - for now, let's assume they're fixed
+    # with Qt 5.13.
+    affected = (qtutils.version_check('5.12', compiled=False) and not
+                qtutils.version_check('5.13', compiled=False))
+    if not affected:
+        return
+
+    log.init.info("Qt version changed, nuking QtWebEngine cache")
+    cache_dir = os.path.join(standarddir.cache(), 'webengine')
+    if os.path.exists(cache_dir):
+        shutil.rmtree(cache_dir)
```

Comment:
It might be nice to move the cache dir to, for example, webengine.old instead of removing. That way, if there are ever any issues or lost data people can go back one migration.